README.md

```diff
@@ -132,6 +139,23 @@ These numbers are based on [scalability tests](https://github.com/kubernetes/kub
 
 Note that if CPU limits are set too low, kube-state-metrics' internal queues will not be able to be worked off quickly enough, resulting in increased memory consumption as the queue length grows. If you experience problems resulting from high memory allocation, try increasing the CPU limits.
 
+#### Horizontal scaling (sharding)
+
+In order to horizontally scale kube-state-metrics some automated sharding capabilities have been implemented. It is configured with the following flags:
+
+* `--shard`
+* `--total-shards`
+
+Sharding is done by taking an md5 sum of the Kubernetes Object's UID and performing a modulo operation on it, with the total number of shards. The configured shard decides whether the object is handled by the respective instance of kube-state-metrics or not. Note that this means all instances of kube-state-metrics even if sharded will have the network traffic and the resource consumption for unmarshaling objects for all objects, not just the ones it is responsible for. To optimize this further, the Kubernetes API would need to support sharded list/watch capabilities. Overall memory consumption should be 1/n th of each shard compared to an unsharded setup. Typically kube-state-metrics needs to be memory and latency optimized in order for it to return its metrics rather quickly to Prometheus.
+
+Sharding should be used carefully, and additional monitoring should be set up in order to ensure that sharding is setup as expected (eg. instances for each shard out of the total shards are configured).
+
+##### Automated sharding
+
+There is also an experimental feature, that allows kube-state-metrics to auto discover its nominal position if it is deployed in a StatefulSet, in order to automatically configure sharding. This is an experimental feature and may be broken or removed without notice.
+
+To enable automated sharding kube-state-metrics must be run by a `StatefulSet` and the pod names and namespace must be handed to the kube-state-metrics process via the `--pod` and `--pod-namespace` flags.
```

Comment:
Instead of kube-state-metrics reaching out to the API-Server to get its index in the total amount of shards, why not derive it from the pod name retrieved via an environment variable?

Reply:
kube-state-metrics needs to know the total number of replicas, that’s what it requests the statefulset for. Detecting it’s nominal within it is done as you said.

Reply:
Right. Sorry about that. And this total # can not be templated from the statefulset status as an environment variable?

Reply:
The downward API only works for data within the pod object, not higher level objects.